sound/soc/sof/ipc4-topology.h

```diff
@@ -349,4 +349,40 @@ struct sof_ipc4_process {
 	struct sof_ipc4_msg msg;
 };
 
+/**
```

Comment:
are we already using this type "base module config + module specific extension"?

Reply:
yes, all bese fw module use this type, for example: 
copier:
```
struct ipc4_copier_module_cfg {
	struct ipc4_base_module_cfg base;

	struct ipc4_audio_format out_fmt;
	uint32_t copier_feature_mask;
	struct ipc4_copier_gateway_cfg  gtw_cfg;
} __attribute__((packed, aligned(4)));
```

peakvol
```
struct ipc4_peak_volume_module_cfg {
	struct ipc4_base_module_cfg base_cfg;
	struct ipc4_peak_volume_config config[0];
} __packed __aligned(8);
```